api/users/views.py

```diff
@@ -436,3 +442,77 @@ def perform_destroy(self, instance):
             if val['id'] in current_institutions:
                 user.remove_institution(val['id'])
         user.save()
+
+class ClaimUser(JSONAPIBaseView, generics.CreateAPIView, UserMixin):
+    permission_classes = (
+        base_permissions.TokenHasScope,
+        ClaimUserPermission
+    )
+
+    required_read_scopes = [CoreScopes.NULL]  # Tokens should not be able to access this
+    required_write_scopes = [CoreScopes.NULL]  # Tokens should not be able to access this
+
+    view_category = 'users'
+    view_name = 'claim-user'
+
+    def _send_claim_email(self, *args, **kwargs):
+        """ This avoids needing to reimplement all of the logic in the sender methods.
+        When v1 is more fully deprecated, those send hooks should be reworked to not
+        rely upon a flask context and placed in utils (or elsewhere).
+
+        bool registered : Indicates which sender to call
+        args, kwargs    : Passed through to mail helpers
+        return          : None
+        """
+        from website.app import app
+        from website.routes import make_url_map
+        try:
+            make_url_map(app)
```

Comment:
Just curious: is there a situation where we would hit this method without the routes being initialized? I guess I'm not sure why this is necessary.

Reply:
It's outside of a flask context, so the first time it's hit (indicated by test behavior, at least).

Reply:
I think the URL map should always be initialized when running the app since we call `init_app` in `manage.py`. Not sure about tests, though. In any case, I don't have a problem keeping this here.

Reply:
We call `init_app` with `routes=False` in `manage.py`, which causes it to not call `make_url_map`.

Reply:
Ah ok, good catch.